documentation/projects/proposals/trust_and_safety/detecting_sensitive_textual_content/20230506-implementation_plan_frontend_blurring_sensitive.md

```diff
@@ -0,0 +1,232 @@
+URL-based malicious action.
+
+#### `sensitive_content_handling`
+
+This is the main, real feature flag that enables the entire feature of being
+able to get (or avoid) mature content and then see or blur it. This flag will
+determine whether the two switches even appear in the sidebar.
+
+It will be set to 'enabled' in development, 'switchable' in staging and
+'disabled' in production till launch, post which it will be 'enabled'
+everywhere.
+
+#### What about the blur toggle?
+
+The second toggle in the mockups that determines whether to blur the sensitive
+results (let's refer to as `blur_sensitive`) is an ephemeral toggle that is
+off + disabled when `fetch_sensitive` is off and enabled only when
+`fetch_sensitive` is on. It is stored temporarily in the state and resets to off
+on a hard refresh.
+
+### Step 3: Updating search store
+
+In this step we will update the search mechanism to add support for the
+`include_sensitive_results` parameter, which supersedes the existing `mature`
+parameter. The list of search results will include mature content if this is set
+to `true`.
+
+Since the API will also send the `sensitivity` field, the types for media items
+will need to be updated to account for that. This will boil down to a boolean
+value of `is_sensitive` for each media item.
+
+### Step 4: Blurring in search results
+
+The approaches to concealment of sensitive content in the search result page is
+specific to the media type. For image results, the image itself is blurred; for
+audio results, the title is blurred.
+
+The cell components for search results, i.e. image cell and two layouts (box and
+row) of the audio track will be updated to automatically blur the right parts of
+the UI, based on the store state of the
+[`blur_sensitive` toggle](#what-about-the-blur-toggle) and the `is_sensitive`
+value for their respective media item.
+
+We are currently going with the CSS blur approach.
+
+### Step 5: Blurring in single result pages
+
+The single result page will be updated to render a safety wall when the media
+item is sensitive. The safety wall will be a full page overlay that will conceal
+the entire content of the page.
+
+It will have a button that will allow the user to proceed to see the content or
+to go to a default page.
+
+There is a difference in the behaviour of this page based on whether the user
+has arrived to it from a search result page or directly. In the former case, the
+preference for the [`blur_sensitive` toggle](#what-about-the-blur-toggle) will
+be carried over to the single result page (as they will be in the same session),
+and if the user has opted to see sensitive content unblurred, the wall will not
+be shown to them.
+
+Care must be taken to not show the content at all unless we have the information
+about the sensitivity of the media item. A flash of clearly visible sensitive
+content before the wall appears will be jarring and will defeat the purpose of
+the wall.
```

Comment:
It might be worth mentioning here explicitly that we _should not render_ the sensitive media until the user opts-in. As opposed to, for example, actually rendering the media in the DOM and rendering the "wall" on top of it.

Reply:
Would it not be better to have it fetched and loaded in the DOM but not render it so that the experience is smoother once the user opts-in? The wall would not be "on-top" of the content but the content could be loaded and `display: none`-ed.

I'm not opposed to the idea of not having it in the DOM before opt-in, that is perfectly reasonable, but I'm wondering if that is indeed the best UX.

Reply:
In the dom, but `display: none`-ed seems fine to me!